config/v1/stable.clusterversion.testsuite.yaml

```diff
@@ -14,3 +14,108 @@ tests:
       kind: ClusterVersion
       spec:
         clusterID: foo
+  - name: Should allow image to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          image: bar
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          image: bar
+  - name: Should allow version to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          version: 4.11.1
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          version: 4.11.1
+  - name: Should allow architecture and version to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+  - name: Should not allow architecture and image to be set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+          version: 4.11.1
+          image: bar
+    expectedError: "cannot set both Architecture and Image"
+  - name: Version must be set if architecture is set
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: ClusterVersion
+      spec:
+        clusterID: foo
+        desiredUpdate:
+          architecture: Multi
+    expectedError: "Version must be set if Architecture is set"
+  onUpdate:
```

Comment:
Ok, I am going to remove this test.

Reply:
I had meant the entire set of onUpdate, none of them seem necessary since they all duplicate previous test cases IIUC. But I see now why you've got two different ones, it's the transitions from image to both and from architecture to both, makes sense, we should add an onCreate version though

Reply:
Both onUpdate tests are covered by the following onCreate test
```
- name: Should not allow image and architecture to be set
    initial: |
      apiVersion: config.openshift.io/v1
      kind: ClusterVersion
      spec:
        clusterID: foo
        desiredUpdate:
          architecture: Multi
          version: 4.11.1
          image: bar
```

Reply:
We have onCreate `Should not allow image and architecture to be set` which checks that both cannot be set so I'm not sure what can be added to onCreate to replicate the two onUpdates since each of them can start with only one of the values but the onCreate cannot (AFAIK).

Reply:
And I originally added the `onUpdate` tests to cover a user doing an `oc patch` for example.

Reply:
Also I can work on this even after the PR is merged. So lets not block the PR to get merged because of this.